From 2cbbefbe53ba07d268e54dc682eadcc022631349 Mon Sep 17 00:00:00 2001
From: Wilson Yang <c_yangw@qca.qualcomm.com>
Date: Tue, 1 Oct 2013 20:11:19 -0700
Subject: [PATCH 1/2] wlan: Cached PMK ID still included after forget AP

implement _del_pmksa/_flush_pmksa in wlan_hdd_cfg80211.c
to support deleting PMKID from the cache when forgetting
the AP from GUI

Change-Id: I6222ee31051b5182ea33f33cd189331bbf719e6f
CRs-fixed: 536894
---
 CORE/HDD/src/wlan_hdd_cfg80211.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CORE/HDD/src/wlan_hdd_cfg80211.c b/CORE/HDD/src/wlan_hdd_cfg80211.c
index 9d6085f..23420bc 100644
--- a/CORE/HDD/src/wlan_hdd_cfg80211.c
+++ b/CORE/HDD/src/wlan_hdd_cfg80211.c
@@ -7447,7 +7447,9 @@ static int wlan_hdd_cfg80211_del_pmksa(struct wiphy *wiphy, struct net_device *d
 
              /*delete the last PMKID cache in CSR*/
              if (eHAL_STATUS_SUCCESS !=
-                 sme_RoamDelPMKIDfromCache(halHandle, pAdapter->sessionId, pmksa->bssid))
+                 sme_RoamDelPMKIDfromCache(halHandle,
+                                           pAdapter->sessionId,
+                                           pBSSID))
              {
                 hddLog(VOS_TRACE_LEVEL_ERROR,"%s: cannot delete PMKSA %d CONTENT.",
                           __func__,PMKIDCacheIndex);
-- 
1.8.2.1

